core/service/transactionPool.ts

```diff
@@ -141,16 +155,42 @@ class TransactionPoolService<T extends object> implements ITransactionPoolServic
 
         if (trs.type === TransactionType.SEND) {
             const asset: IAssetTransfer = <IAssetTransfer>trs.asset;
-            if (this.poolByRecipient.has(asset.recipientAddress) &&
-                this.poolByRecipient.get(asset.recipientAddress).indexOf(trs) !== -1
-            ) {
-                this.poolByRecipient.get(asset.recipientAddress)
-                    .splice(this.poolByRecipient.get(asset.recipientAddress).indexOf(trs), 1);
-            }
+            this.removeOneFromPoolByRecipient(asset.recipientAddress, trs);
         }
+
+        if (trs.type === TransactionType.VOTE) {
```

Comment:
in TransactionType.VOTE or TransactionType.STAKE -> IAssetVote | IAssetStake in type